### DISCUSSION TITLE
Cache actions instead of downloading it each time in self hosted runners

### DISCUSSION THREAD

**Discussion by Puneeth-n** [actions/runner#1157] [Q&A]
Is there a way to cache the actions instead of downloading them each time?

**Comment by TingluoHuang** [comment]
We didn't cache it, so it won't get polluted by other jobs.

**Reply by Puneeth-n** [reply]
@TingluoHuang Is there anyway to configure this option on self hosted runners?

**Reply by j3parker** [reply]
On some level with non-ephemeral runners pollution between jobs is inevitable (actions/checkout will tidy up the workspace to a degree, but there are so many other ways to cause pollution e.g. orphaned processes, if you allow sudo, etc.) but I'm sure there would be non-zero effort to implement this on your end and I respect that.

For our part we use ephemeral runners with a machine image that is based on the virtual-environments repo. We don't care about caching actions that are cloned as part of the job (because after the job the VM goes away) but having a way to pre-cache them in the image would be nice.

It would save us a couple of seconds per job which is minor but not nothing. For us it would be particularly easy because we require our private repos to pull third-party actions through a dependabot-maintained mirror like this: https://github.com/brightspace/third-party-actions so we would know exactly what things we needed to cache :)
